### PR TITLE
Missing Encoding and RequiredProperties when IformFile withOpenApi

### LINKED ISSUE #2625: Incorrect schema generation for IFormFile parameters in Minimal API projects when using WithOpenApi extension method
I'm using .NET 7.0.4 and Swashbuckle.AspNetCore 6.5.0.

Consider the following endpoints:

    app.MapPost("/api/upload", (IFormFile file) =>
    {
        return TypedResults.Ok(file.FileName);
    });
    
    app.MapPost("/api/upload_with_openapi", (IFormFile file) =>
    {
        return TypedResults.Ok(file.FileName);
    })
    .WithOpenApi();

The first one generates the following schema:

    schema": {
      "required": [
        "file"
      ],
      "type": "object",
      "properties": {
        "file": {
          "type": "string",
          "format": "binary"
        }
      }
    },

While the second, using `WithOpenApi`, produces this:

    "schema": {
      "type": "string",
      "format": "binary"
    },

This causes an exception when calling the `/api/upload_with_openapi` endpoint:

```
Microsoft.AspNetCore.Http.BadHttpRequestException: Required parameter "IFormFile file" was not provided from form file.
       at lambda_method3(Closure, Object, HttpContext, Object)
       at Microsoft.AspNetCore.Http.RequestDelegateFactory.<>c__DisplayClass92_2.<<HandleRequestBodyAndCompileRequestDelegateForForm>b__2>d.MoveNext()
    --- End of stack trace from previous location ---
       at Microsoft.AspNetCore.Routing.EndpointMiddleware.<Invoke>g__AwaitRequestTask|6_0(Endpoint endpoint, Task requestTask, ILogger logger)
       at Swashbuckle.AspNetCore.SwaggerUI.SwaggerUIMiddleware.Invoke(HttpContext httpContext)
       at Swashbuckle.AspNetCore.Swagger.SwaggerMiddleware.Invoke(HttpContext httpContext, ISwaggerProvider swaggerProvider)
       at Microsoft.AspNetCore.Diagnostics.DeveloperExceptionPageMiddlewareImpl.Invoke(HttpContext context)
```

You can find a minimal repro here: https://github.com/marcominerva/FileUploadIssue

_Original issue posted at https://github.com/dotnet/aspnetcore/issues/47526_

### PR DESCRIPTION
**Pull request by jgarciadelanoceda**
Finally fixes #2625

### CODE REVIEW

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs [hunk 1]
```diff
@@ -442,25 +443,30 @@ private OpenApiOperation GenerateOpenApiOperationFromMetadata(ApiDescription api
...
-                    static OpenApiSchema GenerateSchemaIncludingFormFile(ApiParameterDescription apiParameterDescription, OpenApiSchema generatedSchema)
+                    static OpenApiSchema GenerateSchemaIncludingFormFile(ApiParameterDescription apiParameterDescription, OpenApiSchema generatedSchema, OpenApiMediaType mediaType)
                     {
-                        if (generatedSchema.Reference is null
-                            && ((generatedSchema.Type == "string" && generatedSchema.Format == "binary") || (generatedSchema.Type == "array" && generatedSchema.Items.Format == "binary")))
+                        if (generatedSchema.Reference is null && apiParameterDescription.IsFromForm())
                         {
-                            return new OpenApiSchema()
+                            mediaType.Encoding.Add(apiParameterDescription.Name, new OpenApiEncoding { Style = ParameterStyle.Form });
+                            if((generatedSchema.Type == "string" && generatedSchema.Format == "binary")
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:457]:
```suggestion
                            if ((generatedSchema.Type == "string" && generatedSchema.Format == "binary")
```

File: src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs [hunk 2]
```diff
@@ -442,25 +443,30 @@ private OpenApiOperation GenerateOpenApiOperationFromMetadata(ApiDescription api
...
                                 {
-                                    [apiParameterDescription.Name] = generatedSchema
-                                }
-                            };
+                                    Type = "object",
+                                    Properties = new Dictionary<string, OpenApiSchema>()
+                                    {
+                                        [apiParameterDescription.Name] = generatedSchema
+                                    },
+                                    Required = apiParameterDescription.IsRequired ?  new SortedSet<string>() { apiParameterDescription.Name} : null
```
Comment (martincostello) [inline src/Swashbuckle.AspNetCore.SwaggerGen/SwaggerGenerator/SwaggerGenerator.cs:467]:
```suggestion
                                    Required = apiParameterDescription.IsRequired ? new SortedSet<string>() { apiParameterDescription.Name } : null
```

**Review by martincostello — COMMENTED** [review]
Did you check that this behaves correctly in Swagger-UI itself as per the comment we re-opened the issue for?

**Comment by jgarciadelanoceda** [comment]
Yes I did see it on Swagger-Ui :)

**Review by martincostello — APPROVED** [review]

**Pull request merged**